### PR TITLE
Introduce handshake command + tighter ping UI

### LINKED ISSUE #49: [BUG] - `cardano-cli ping --unixsock` not sending pings
## Description

The `cardano-cli ping --unixsock` is not sending pings. The behavior doesn't seem to correspond with the command name as is potentially confusing for end user.

```
$ cardano-cli ping --count 5 --unixsock /home/martink/Source/repos/cardano-node/state-cluster0/bft1.socket --magic 42 --json
/home/martink/Source/repos/cardano-node/state-cluster0/bft1.socket network rtt: 0.000
/home/martink/Source/repos/cardano-node/state-cluster0/bft1.socket handshake rtt: 0.000188265s
/home/martink/Source/repos/cardano-node/state-cluster0/bft1.socket Negotiated version NodeToClientVersionV16 42
] }
```

Also notice the confusing json output.

Tested with cardano-cli 8.2.1 on Linux.

---

**[edit]** after discussions with @mkoura and @coot. We will:

* Introduce a new `handshake` command that works for both node-to-client and node-to-node.
* Remove the `--unixsock` option from `ping`

---

**[edit]** after review by @Jimbo4350 [here](https://github.com/input-output-hk/cardano-cli/pull/468#discussion_r1402195666), we finally decided to:

* Introduce a new `--handshake-only` flag to `ping`
* Make `-u/--unixsock` work only with `--handshake-only` (i.e. `--unixsock` is conditioned by presence of `--handshake-only` is specified)
* Deprecate `-Q/--query-versions` in `ping`. `-Q` should preferably be used with `--handshake-only`.

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Remove the `-u/--unixsock` option from `cardano-cli ping`,
    because it wasn't performing ping; only handshake.

    Add the `cardano-cli handshake` commands,
    that support both distant hosts (`-h/--host`, `-p/--port`)
    and local hosts (`-u/--unixsock`).
    Like its name suggests, `cardano-cli handshake` only does the handshake part of `cardano-cli ping`.
    
    Deprecate `cardano-cli ping`'s `-Q,--query-versions` flags.
    You should use the same flag with `cardano-cli handshake` instead.
    
    TLDR: if you were using `cardano-cli ping -Q`, use `cardano-cli handshake` instead.
    And don't pretend `cardano-cli ping` works on unix sockets.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

* Fixes https://github.com/input-output-hk/cardano-cli/issues/49
* UI was validated by @coot and @mkoura 

# How to trust this PR

* Execute `cardano-cli handshake`, witness it works
* Execute `cardano-cli ping -u SOCKET`, witness it doesn't exist anymore.

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- NA New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [X] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Run/HandShake.hs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import qualified Cardano.CLI.Run.Ping as Ping
+import qualified Cardano.Network.Ping as CNP
+
+import           Control.Applicative ((<|>))
+import           Data.Word (Word32)
+import qualified Options.Applicative as Opt
+import qualified Prettyprinter as PP
+
+data HandShakeCmd = HandShakeCmd
+  { endPoint        :: !Ping.EndPoint
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Run/HandShake.hs:24]: This shouldn't be a separate command. It should live under the ping command as something like `handshake-only`.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

File: cardano-cli/src/Cardano/CLI/Run/HandShake.hs
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/Run/HandShake.hs:1]: handshake is a single word, so the module should be named `Handshake`

File: cardano-cli/src/Cardano/CLI/Run/HandShake.hs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeApplications #-}
+
+{- HLINT ignore "Move brackets to avoid $" -}
+
+module Cardano.CLI.Run.HandShake
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/Run/HandShake.hs:7]: Handshake is a single word, so the module should be named `Handshake`

```suggestion
module Cardano.CLI.Run.Handshake
```

File: cardano-cli/src/Cardano/CLI/Run/HandShake.hs (hunk 1 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/Run/HandShake.hs:24]: We've settled for `cardano-cli ping --handshake-only`  [here on Slack](https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1701360655094909?thread_ts=1701253501.603859&cid=C031GFLC3DF).

**Comment by smelc** [comment]
Will likely be superseded by #524 in the end
